### DISCUSSION TITLE
xiao_esp32s3 - simple application builds on v3.6, breaks on main

### DISCUSSION THREAD

**Discussion by OOHehir** [zephyrproject-rtos/zephyr#71841] [Q&A]
Hello all,
I've created a [PR](https://github.com/zephyrproject-rtos/zephyr/pull/71513) to add a sample to demonstrate the touch peripheral on the ESP32 device. It builds successfully on Zephyr v3.6 but currently on main shows errors as listed below. Is this a known issue?

Thanks for all your hard work!

```/opt/zephyr-sdk-0.16.5-1/xtensa-espressif_esp32s3_zephyr-elf/bin/../lib/gcc/xtensa-espressif_esp32s3_zephyr-elf/12.2.0/../../../../xtensa-espressif_esp32s3_zephyr-elf/bin/ld.bfd: zephyr/drivers/input/libdrivers__input.a(input_esp32_touch_sensor.c.obj):(.literal.esp32_touch_sensor_init+0x0): undefined reference to `touch_sensor_channel_io_map'
/opt/zephyr-sdk-0.16.5-1/xtensa-espressif_esp32s3_zephyr-elf/bin/../lib/gcc/xtensa-espressif_esp32s3_zephyr-elf/12.2.0/../../../../xtensa-espressif_esp32s3_zephyr-elf/bin/ld.bfd: zephyr/drivers/input/libdrivers__input.a(input_esp32_touch_sensor.c.obj):(.literal.esp32_touch_sensor_init+0x1c): undefined reference to `touch_hal_init'
/opt/zephyr-sdk-0.16.5-1/xtensa-espressif_esp32s3_zephyr-elf/bin/../lib/gcc/xtensa-espressif_esp32s3_zephyr-elf/12.2.0/../../../../xtensa-espressif_esp32s3_zephyr-elf/bin/ld.bfd: zephyr/drivers/input/libdrivers__input.a(input_esp32_touch_sensor.c.obj):(.literal.esp32_touch_sensor_init+0x20): undefined reference to `rtc_gpio_init'
/opt/zephyr-sdk-0.16.5-1/xtensa-espressif_esp32s3_zephyr-elf/bin/../lib/gcc/xtensa-espressif_esp32s3_zephyr-elf/12.2.0/../../../../xtensa-espressif_esp32s3_zephyr-elf/bin/ld.bfd: zephyr/drivers/input/libdrivers__input.a(input_esp32_touch_sensor.c.obj):(.literal.esp32_touch_sensor_init+0x24): undefined reference to `rtc_gpio_set_direction'
/opt/zephyr-sdk-0.16.5-1/xtensa-espressif_esp32s3_zephyr-elf/bin/../lib/gcc/xtensa-espressif_esp32s3_zephyr-elf/12.2.0/../../../../xtensa-espressif_esp32s3_zephyr-elf/bin/ld.bfd: zephyr/drivers/input/libdrivers__input.a(input_esp32_touch_sensor.c.obj):(.literal.esp32_touch_sensor_init+0x28): undefined reference to `rtc_gpio_pulldown_dis'
/opt/zephyr-sdk-0.16.5-1/xtensa-espressif_esp32s3_zephyr-elf/bin/../lib/gcc/xtensa-espressif_esp32s3_zephyr-elf/12.2.0/../../../../xtensa-espressif_esp32s3_zephyr-elf/bin/ld.bfd: zephyr/drivers/input/libdrivers__input.a(input_esp32_touch_sensor.c.obj):(.literal.esp32_touch_sensor_init+0x2c): undefined reference to `rtc_gpio_pullup_dis'
/opt/zephyr-sdk-0.16.5-1/xtensa-espressif_esp32s3_zephyr-elf/bin/../lib/gcc/xtensa-espressif_esp32s3_zephyr-elf/12.2.0/../../../../xtensa-espressif_esp32s3_zephyr-elf/bin/ld.bfd: zephyr/drivers/input/libdrivers__input.a(input_esp32_touch_sensor.c.obj):(.literal.esp32_touch_sensor_init+0x30): undefined reference to `touch_hal_config'
/opt/zephyr-sdk-0.16.5-1/xtensa-espressif_esp32s3_zephyr-elf/bin/../lib/gcc/xtensa-espressif_esp32s3_zephyr-elf/12.2.0/../../../../xtensa-espressif_esp32s3_zephyr-elf/bin/ld.bfd: zephyr/drivers/input/libdrivers__input.a(input_esp32_touch_sensor.c.obj):(.literal.esp32_touch_sensor_init+0x38): undefined reference to `touch_hal_filter_set_config'
/opt/zephyr-sdk-0.16.5-1/xtensa-espressif_esp32s3_zephyr-elf/bin/../lib/gcc/xtensa-espressif_esp32s3_zephyr-elf/12.2.0/../../../../xtensa-espressif_esp32s3_zephyr-elf/bin/ld.bfd: zephyr/drivers/input/libdrivers__input.a(input_esp32_touch_sensor.c.obj): in function `esp32_touch_sensor_init':```

**Comment by kartben** [comment]
Sorry for asking but you did run `west update`, yes? Espressif recently updated their HAL (#68760) so the issue is probably somewhere there.
cc @sylvioalves

**Comment by OOHehir** [comment]
@kartben cc @sylvioalves No bother. The procedure I'm following is:
from zephyr directory:
```
git checkout v3.6-branch
Updating files: 100% (24585/24585), done.
Switched to branch 'v3.6-branch'
Your branch is up to date with 'origin/v3.6-branch'.
```
I'm building an independent application & so from application directory:
```
source ../zephyrproject/.venv/bin/activate
source ../zephyrproject/zephyr/zephyr-env.sh
west update
west blobs fetch hal_espressif
west build -b xiao_esp32s3 ./ -p always
```
... successful build.

For main, similar enough:
```
git checkout main
Updating files: 100% (24585/24585), done.
Switched to branch 'main'
Your branch is up to date with 'origin/main'.
```
from application directory:
```
source ../zephyrproject/.venv/bin/activate
source ../zephyrproject/zephyr/zephyr-env.sh
west update
west blobs fetch hal_espressif 
(significant changes ):
Fetching blob hal_espressif: /home/projects/zephyrproject/modules/hal/espressif/zephyr/blobs/lib/esp32c3/libbtdm_app.a
Fetching blob hal_espressif: /home/projects/zephyrproject/modules/hal/espressif/zephyr/blobs/lib/esp32s3/libbtdm_app.a
Fetching blob hal_espressif: /home/projects/zephyrproject/modules/hal/espressif/zephyr/blobs/lib/esp32/libbtdm_app.a
Fetching blob hal_espressif: /home/projects/zephyrproject/modules/hal/espressif/zephyr/blobs/lib/esp32/libcore.a
Fetching blob hal_espressif: /home/projects/zephyrproject/modules/hal/espressif/zephyr/blobs/lib/esp32/libmesh.a
Fetching blob hal_espressif: /home/projects/zephyrproject/modules/hal/espressif/zephyr/blobs/lib/esp32/libsmartconfig.a
Fetching blob hal_espressif: /home/projects/zephyrproject/modules/hal/espressif/zephyr/blobs/lib/esp32/libnet80211.a
Fetching blob hal_espressif: /home/projects/zephyrproject/modules/hal/espressif/zephyr/blobs/lib/esp32/libpp.a
Fetching blob hal_espressif: /home/projects/zephyrproject/modules/hal/espressif/zephyr/blobs/lib/esp32/libespnow.a
Fetching blob hal_espressif: /home/projects/zephyrproject/modules/hal/espressif/zephyr/blobs/lib/esp32/libwapi.a
Fetching blob hal_espressif: /home/projects/zephyrproject/modules/hal/espressif/zephyr/blobs/lib/esp32s2/libcore.a
Fetching blob hal_espressif: /home/projects/zephyrproject/modules/hal/espressif/zephyr/blobs/lib/esp32s2/libmesh.a
Fetching blob hal_espressif: /home/projects/zephyrproject/modules/hal/espressif/zephyr/blobs/lib/esp32s2/libsmartconfig.a
Fetching blob hal_espressif: /home/projects/zephyrproject/modules/hal/espressif/zephyr/blobs/lib/esp32s2/libnet80211.a
Fetching blob hal_espressif: /home/projects/zephyrproject/modules/hal/espressif/zephyr/blobs/lib/esp32s2/libpp.a
Fetching blob hal_espressif: /home/projects/zephyrproject/modules/hal/espressif/zephyr/blobs/lib/esp32s2/libespnow.a
Fetching blob hal_espressif: /home/projects/zephyrproject/modules/hal/espressif/zephyr/blobs/lib/esp32s2/libwapi.a
Fetching blob hal_espressif: /home/projects/zephyrproject/modules/hal/espressif/zephyr/blobs/lib/esp32c3/libcore.a
Fetching blob hal_espressif: /home/projects/zephyrproject/modules/hal/espressif/zephyr/blobs/lib/esp32c3/libmesh.a
Fetching blob hal_espressif: /home/projects/zephyrproject/modules/hal/espressif/zephyr/blobs/lib/esp32c3/libsmartconfig.a
Fetching blob hal_espressif: /home/projects/zephyrproject/modules/hal/espressif/zephyr/blobs/lib/esp32c3/libnet80211.a
Fetching blob hal_espressif: /home/projects/zephyrproject/modules/hal/espressif/zephyr/blobs/lib/esp32c3/libpp.a
Fetching blob hal_espressif: /home/projects/zephyrproject/modules/hal/espressif/zephyr/blobs/lib/esp32c3/libespnow.a
Fetching blob hal_espressif: /home/projects/zephyrproject/modules/hal/espressif/zephyr/blobs/lib/esp32c3/libwapi.a
Fetching blob hal_espressif: /home/projects/zephyrproject/modules/hal/espressif/zephyr/blobs/lib/esp32s3/libcore.a
Fetching blob hal_espressif: /home/projects/zephyrproject/modules/hal/espressif/zephyr/blobs/lib/esp32s3/libmesh.a
Fetching blob hal_espressif: /home/projects/zephyrproject/modules/hal/espressif/zephyr/blobs/lib/esp32s3/libsmartconfig.a
Fetching blob hal_espressif: /home/projects/zephyrproject/modules/hal/espressif/zephyr/blobs/lib/esp32s3/libnet80211.a
Fetching blob hal_espressif: /home/projects/zephyrproject/modules/hal/espressif/zephyr/blobs/lib/esp32s3/libpp.a
Fetching blob hal_espressif: /home/projects/zephyrproject/modules/hal/espressif/zephyr/blobs/lib/esp32s3/libespnow.a
Fetching blob hal_espressif: /home/projects/zephyrproject/modules/hal/espressif/zephyr/blobs/lib/esp32s3/libwapi.a
Fetching blob hal_espressif: /home/projects/zephyrproject/modules/hal/espressif/zephyr/blobs/lib/esp32s2/libphy.a
Fetching blob hal_espressif: /home/projects/zephyrproject/modules/hal/espressif/zephyr/blobs/lib/esp32c3/libphy.a
Fetching blob hal_espressif: /home/projects/zephyrproject/modules/hal/espressif/zephyr/blobs/lib/esp32c3/libbtbb.a
Fetching blob hal_espressif: /home/projects/zephyrproject/modules/hal/espressif/zephyr/blobs/lib/esp32s3/libphy.a
Fetching blob hal_espressif: /home/projects/zephyrproject/modules/hal/espressif/zephyr/blobs/lib/esp32s3/libbtbb.a
```
Due to changes I then use an overlay:
```
xiao_esp32s3_esp32s3_procpu.overlay
```
& build with:
```
west build -b xiao_esp32s3/esp32s3/procpu ./ -p always
```
... unsuccessful build

**Comment by sylvioalves** [comment]
I have just tried `rm -rf build; west build -b xiao_esp32s3 samples/hello_world/` and it worked. Can you check with `hello_world` to make sure whether it works?

**Comment by OOHehir** [comment]
@sylvioalves I can probably do something better..
My prj.conf is fairly simple
```
CONFIG_INPUT=y
CONFIG_INPUT_ESP32_TOUCH_SENSOR=y
```
If I comment out CONFIG_INPUT_ESP32_TOUCH_SENSOR (& remove any associated dependencies in the application) the build is successful for branch main.

**Comment by sylvioalves** [comment]
Tested building with `rm -rf build; west build -b xiao_esp32s3/esp32s3/procpu samples/hello_world/ -DCONFIG_INPUT=y -DCONFIG_INPUT_ESP32_TOUCH_SENSOR=y` and see no issues. Will close this.

**Comment by RomainPelletant** [comment]
I am facing the same issue on main branch (esp32-wroom)
Environment (same results in both): 
* zephyrprojectrtos/ci:v0.26.12 (zephyr-sdk-0.16.6)
* MacOS (zephyr-sdk-0.16.8)

The CMakeLists.txt in components/hal seems not called
Succeed to compile by adding manually the following lines in hal/espressif/zephyr/esp32/CMakeLists.txt

```
zephyr_sources(
    ../../components/hal/touch_sensor_hal.c
    ../../components/hal/${CONFIG_SOC_SERIES}/touch_sensor_hal.c
    ../../components/hal/${CONFIG_SOC_SERIES}/gpio_hal_workaround.c
    ../../components/driver/gpio/rtc_io.c
    ../../components/hal/rtc_io_hal.c
    ../../components/soc/${CONFIG_SOC_SERIES}/touch_sensor_periph.c
    )
```
